app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -42,6 +42,11 @@ cat << EOF > $SSL_CA_CERT
 <%= foreman_server_ca_cert %>
 EOF
 
+cleanup_and_exit() {
+  rm -f $SSL_CA_CERT
```

Comment:
this doesn't cleanup `KATELLO_SERVER_CA_CERT` or roll-back the changes to `rhsm.conf`

is that intentional?

Reply:
(probably, because it's *also* called in the success case, but I think we should be nicer in the error case)

Reply:
I agree with this. Today, on uninstall users set back to `rhsm.conf.bak`. If this fails, it would be nice to restore them to whatever they had originally so they are not left in a broken state. That is, on failure restore the rhsm.conf.bak to rhsm.conf. Use cases:

1) User is moving a host from being registered to Red Hat CDN to a Foreman, something fails, they would get reverted and remain connected and operational
2) User is moving a host from one Foreman to another Foreman, this would prevent the host from being in limbo if it failed

Reply:
> this doesn't cleanup KATELLO_SERVER_CA_CERT or roll-back the changes to rhsm.conf
> 
> is that intentional?

But do we want to remove `KATELLO_SERVER_CA_CERT`? If we remove the cert  then all communication between `subscription-manager` & foreman will fail with SSL error, right?

> Today, on uninstall users set back to rhsm.conf.bak

We should do the backup of the `rhsm.conf` that's for sure.

> If this fails, it would be nice to restore them to whatever they had originally so they are not left in a broken state. That is, on failure restore the rhsm.conf.bak to rhsm.conf.

Is this something what we do right now with `katello_ca_consumer.rpm`? I mean, if registration with `rpm` fails, does it do all the rollback steps automatically or users have to do them manually?

Because implementing rollback feature (while catching all the possible fail scenarios) won't be easy.
We need to cleanup methods:
* When script is successful
* When script failed

IMHO this is out of the scope of this PR, I would suggest to do it in separate PR and not block this one

Reply:
> But do we want to remove `KATELLO_SERVER_CA_CERT`? If we remove the cert then all communication between `subscription-manager` & foreman will fail with SSL error, right?

This and the rhsm.conf roll back is tricky, as today we do this if the RPM is removed (https://github.com/theforeman/puppet-foreman_proxy_content/blob/master/lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb#L203). But I do not see us doing it if the re-configure script fails. I would agree that could be left for a follow up given this does reflect current state.